funcx_endpoint/funcx_endpoint/executors/high_throughput/messages.py

```diff
@@ -188,12 +190,19 @@ def unpack(cls, msg):
         endpoint_id = str(uuid.UUID(bytes=msg[:16]))
         msg = msg[16:]
         jsonified = msg.decode("ascii")
-        ep_status, task_statuses = json.loads(jsonified)
+        ep_status, statuses = json.loads(jsonified)
+        task_statuses = {}
+        for tid, ts in statuses.items():
+            task_statuses[tid] = TaskTransition(
+                timestamp=ts["timestamp"], actor=ts["actor"], state=ts["state"]
+            )
         return cls(endpoint_id, ep_status, task_statuses)
 
     def pack(self):
-        # TODO: do we want to do better than JSON?
-        jsonified = json.dumps([self.ep_status, self.task_statuses])
+        statuses = {}
+        for tid, status in self.task_statuses.items():
+            statuses[tid] = status.to_dict()
+        jsonified = json.dumps([self.ep_status, statuses])
```

Comment:
Does this get cleaner if the aforementioned data structure change is a good suggestion?